src/app/QuickStarts/QuickStartDrawer.tsx

```diff
@@ -0,0 +1,139 @@
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import build from '@app/build.json';
+import { LoadingView } from '@app/LoadingView/LoadingView';
+import { allQuickStarts } from '@app/QuickStarts/all-quickstarts';
+import { SessionState } from '@app/Shared/Services/Login.service';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { FeatureLevel } from '@app/Shared/Services/Settings.service';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import {
+  QuickStartContext,
+  QuickStartDrawer,
+  useLocalStorage,
+  useValuesForQuickStartContext,
+} from '@patternfly/quickstarts';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { distinctUntilChanged } from 'rxjs';
+import GenericQuickStart from './quickstarts/generic-quickstart';
+
+const LINK_LABEL = "[\\d\\w\\s-()$!&']+"; // has extra &' in matcher
+const HIGHLIGHT_ACTIONS = ['highlight']; // use native quickstarts highlight markdown extension
+const SELECTOR_ID = `[\\w-&]+`; // has extra &'
+
+// [linkLabel]{{action id}}
+const HIGHLIGHT_REGEXP = new RegExp(`\\[(${LINK_LABEL})]{{(${HIGHLIGHT_ACTIONS.join('|')}) (${SELECTOR_ID})}}`, 'g');
+export interface GlobalQuickStartDrawerProps {
+  children: React.ReactNode;
+}
+
+export const GlobalQuickStartDrawer: React.FC<GlobalQuickStartDrawerProps> = ({ children }) => {
+  const { i18n } = useTranslation();
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const [activeQuickStartID, setActiveQuickStartID] = useLocalStorage('quickstartId', '');
+  const [allQuickStartStates, setAllQuickStartStates] = useLocalStorage('quickstarts', {});
+
+  React.useEffect(() => {
+    addSubscription(
+      context.settings
+        .featureLevel()
+        .pipe(distinctUntilChanged())
+        .subscribe((level) => {
+          if (level === FeatureLevel.DEVELOPMENT) {
+            if (!allQuickStarts.includes(GenericQuickStart)) {
+              allQuickStarts.push(GenericQuickStart);
```

Comment:
Ideally, exported variables should be read-only.

Above, it is actually quite unsafe since somehow this array is updated before quickstart finishes initiates. Try wrapping this callback in `setTimeout` with 500ms, set to DEV Mode, notice the generic quickstart is not showing up.

Is there anyway we can dynamically update the total list of quickstarts?

Reply:
If possible, we can add a metadata to the quickstart object to hold the featureLevel, then call `useFeatureLevel` and use the currect level to filter out quickstarts.